e2e/test/scenarios/dashboard-cards/dashboard-card-reproductions.cy.spec.js

```diff
@@ -1158,7 +1158,7 @@ describe("issue 31628", () => {
         scalarContainer.then($element => assertIsEllipsified($element[0]));
         scalarContainer.realHover();
 
-        popover().findByText("18,760").should("exist");
+        cy.findByRole("tooltip").findByText("18,760").should("exist");
```

Comment:
Nit:
`should("exist")` is a redundant assertion in most cases. Do not change this here because you'll have to wait for the full new CI cycle PLUS we're doing this all over the place. If we don't want to rely on the implicit assertions (if it finds something by text, it obviously exists!), we should at least use `should("be.visible")`

A different approach would be to say `tooltip.should("have.text", "foo")` or `tooltip.should("contain", "foo")`

```suggestion
        cy.findByRole("tooltip").findByText("18,760").should("exist");
```

Reply:
> should("exist") is a redundant assertion in most cases. 
> (if it finds something by text, it obviously exists!)

I always add these assertions explicitly because to me a query without explicit assertion looks like incomplete code - it's not always easy to tell whether it's a leftover code or an actual assertion. Plus it reads like natural language.

Reply:
I get that.
And I'm perfectly fine with NOT using implicit assertions, but that's why I gave you **three** alternative assertions which are even more specific than "Should exist".